src/app/data-explorer/querybuilder/querybuilder.component.html

```diff
@@ -0,0 +1,272 @@
+<!--
+Copyright 2022 University of Oxford
+and Health and Social Care Information Centre, also known as NHS Digital
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+-->
+<div class="query-builder">
+  <p>{{ title }}</p>
+  <textarea readonly class="output">{{ query | jql }}</textarea>
+  <div class="toggle-query-builder-div">
+    <button
+      class="toggle-query-builder-button"
+      type="button"
+      mat-button
+      color="primary"
+      (click)="toggleQueryBuilder()"
+    >
+      <mat-icon>{{ qbController.toggleQueryBuilderIcon }}</mat-icon>
+      {{ qbController.toggleQueryBuilderText }}
+    </button>
+  </div>
+  <!--<div *ngIf="qbController.showQueryBuilder">-->
+  <div class="box" [class.opened]="qbController.showQueryBuilder">
+    <mat-card class="mdm-data-element-card">
+      <query-builder
+        [(ngModel)]="query"
+        [config]="config"
+        [operatorMap]="operatorMap"
+        [allowRuleset]="allowRuleset"
+      >
+        <!-- Add/Remove rule and ruleset buttons -->
+        <ng-container
+          *queryButtonGroup="
+            let ruleset;
+            let addRule = addRule;
+            let addRuleSet = addRuleSet;
+            let removeRuleSet = removeRuleSet
+          "
+        >
+          <button
+            class="add-rule-button"
+            type="button"
+            mat-button
+            color="primary"
+            (click)="addRule()"
+          >
+            <mat-icon>add</mat-icon>
+            Rule
+          </button>
+          <button
+            class="add-rule-button"
+            type="button"
+            mat-button
+            color="primary"
+            *ngIf="addRuleSet"
+            (click)="addRuleSet()"
+          >
+            <mat-icon>add_circle_outline</mat-icon>
+            Rule Set
+          </button>
+          <button
+            type="button"
+            mat-button
+            color="accent"
+            *ngIf="removeRuleSet"
+            (click)="removeRuleSet()"
+          >
+            <mat-icon>remove_circle_outline</mat-icon>
+          </button>
+        </ng-container>
+        <!--  -->
```

Comment:
The comment was to identify what part of the query builder the following code rendered. I had not been able to work it out. I now have and have removed the two sections that had empty comments. These sections were:
1. An arrow for showing and hiding rulesets. I have not turned on the feature to collapse rulesets since I didn't think it improved the user experience and when I did turn it on it didn't work properly.
2. Entity selection. In the demo it shows that an extra dropdown box can be added before the field that is being selected to filter the field list. I don't think we have a good use for this mode, so I have removed the supporting code.

If we decide to use these features, then the code I have removed can always be reinstated.